internal/pkg/consensus/signing_test.go

```diff
@@ -0,0 +1,135 @@
+package consensus
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	tf "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
+	vmaddr "github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
+)
+
+type fakeStateView struct {
+	keys map[address.Address]address.Address
+}
+
+func (f *fakeStateView) AccountSignerAddress(_ context.Context, a address.Address) (address.Address, error) {
+	if a.Protocol() == address.SECP256K1 || a.Protocol() == address.BLS {
+		return a, nil
+	}
+	resolved, ok := f.keys[a]
+	if !ok {
+		return address.Undef, fmt.Errorf("not found")
+	}
+	return resolved, nil
+
+}
+
+func TestSignMessageOk(t *testing.T) {
+	tf.UnitTest(t)
+	ctx := context.Background()
+
+	idAddress := vmaddr.RequireIDAddress(t, 1)
+	ms, kis := types.NewMockSignersAndKeyInfo(1)
+	keyAddr, err := kis[0].Address()
+	require.NoError(t, err)
+
+	{
+		v := NewSignatureValidator(&fakeStateView{}) // No resolution needed.
+		msg := types.NewMeteredMessage(keyAddr, keyAddr, 1, types.ZeroAttoFIL, builtin.MethodSend, nil, types.NewGasPrice(0), 0)
+		smsg, err := types.NewSignedMessage(*msg, ms)
+		require.NoError(t, err)
+		assert.NoError(t, v.ValidateMessageSignature(ctx, smsg))
+	}
+	{
+		// Use ID address in message but sign with corresponding key address.
+		state := &fakeStateView{keys: map[address.Address]address.Address{
+			idAddress: keyAddr,
+		}}
+		v := NewSignatureValidator(state)
+		msg := types.NewMeteredMessage(idAddress, idAddress, 1, types.ZeroAttoFIL, builtin.MethodSend, nil, types.NewGasPrice(0), 0)
+		msgData, err := msg.Marshal()
+		require.NoError(t, err)
+		sig, err := ms.SignBytes(msgData, keyAddr)
+		require.NoError(t, err)
+		smsg := &types.SignedMessage{
+			Message:   *msg,
+			Signature: sig,
+		}
+
+		assert.NoError(t, v.ValidateMessageSignature(ctx, smsg))
+	}
+}
+
+// Signature is valid but signer does not match From Address.
+func TestBadFrom(t *testing.T) {
```

Comment:
It would be nice to see a test case exercising the id address translation similar to the one above to verify.